support/util/imagemetadata.go

```diff
@@ -387,6 +387,19 @@ func GetRegistryOverrides(ctx context.Context, ref reference.DockerImageReferenc
 		return nil, false, fmt.Errorf("failed to parse mirror image reference %q: %w", source, err)
 	}
 
+	// If the root registry is provided, once the image is parsed, it returns the registry in the Name field
```

Comment:
Maybe this section should be its own function to simplify `GetRegistryOverrides` further. IMO, we should refactor `GetRegistryOverrides` because of how complex it has gotten.

Reply:
It seems like we should at least make this section of new code less complex given it has compounded and nested if statements.

Reply:
Refactored, let me know if you have any concern